enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -133,8 +134,30 @@ protected void addInitContainer(PodTemplateSpecBuilder builder, JSONObject initC
         String initContainerAnnotation = builder.buildMetadata().getAnnotations().get(INIT_CONTAINER_ANNOTATION);
         JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
                 : new JSONArray(initContainerAnnotation);
-        initContainers.put(initContainer);
 
+        // lets avoid duplicates being added
+        boolean duplicate = false;
+        int length = initContainers.length();
+        for (int i = 0; i < length; i++) {
+            JSONObject obj = initContainers.getJSONObject(i);
+            if (duplicateInitcontainer(obj, initContainer)) {
+                duplicate = true;
+            }
+
+        }
+        if (!duplicate) {
+            initContainers.put(initContainer);
+        }
         builder.editMetadata().addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString()).endMetadata();
     }
+
+    private boolean duplicateInitcontainer(JSONObject a, JSONObject b) {
+        Object name1 = a.get("name");
+        Object name2 = b.get("name");
+        if (Objects.equal(name1, name2)) {
+            log.warn("Found duplicate init containers with names " + name1 + " so ignoring " + b);
```

Comment:
Yes code reviews slow things down, superficially at least. The extent they slow things down is dependent on the scope of the change. @rhuss especially is extremely quick at giving feedback so shouldn't slow down the f-m-p significantly but should prevent unknown technical debt being introduced without tracking for when it needs to be cleaned up later. We are also trying to get more maintainers that would help speed up code reviews by sharing the load.

The reason we're spending time discussing this change is not the change itself, but rather the hack a fix, self-merge, review and fix properly later (maybe). While this has a perception of being faster for you, what's actually happening is that other people (@rhuss in this case) are having to spend much longer undoing the changes to make the code base predictable, maintainable and supportable.

In lieu of automated testing for dependency/plugin upgrades, perhaps manual testing before merging upgrades would be prudent?